library/public/types.h

```diff
@@ -62,6 +79,16 @@ struct mesh_t
    */
   F3D_EXPORT std::pair<bool, std::string> isValid() const;
 };
+
+enum class option_types
+{
+  _bool,
+  _int,
+  _double,
+  _ratio,
+  _string,
+  _double_vector
+};
```

Comment:
Can we move that in `options_tools.h.in`, I'm not comfortable with the idea of exposing that to the user.